src/types.ts

```diff
@@ -49,7 +49,11 @@ export interface BidsDto {
   tokenId: string;
 }
 
-export interface Bid extends BidParams {
+export interface Cancellable {
```

Comment:
Instead consider something like

```
export interface CanceledBid extends Bid {
  cancelDate: number;
}
```

then you can take in a `Bid | null` from the db in `api.ts > /bid/cancel` and turn it into a `CancelledBid` by giving the `cancelDate` a value, removing the need for ambiguity checks with `??`

Reply:
I was considering that, but was also worried about in the future if Bid is ever extended any further, then it would have to extend CanceledBid instead or else canceledBid might be orphaned

Reply:
It wouldn't matter what happens to `Bid` in the future if `CanceledBid` extends from it like this because the `CanceledBid` will always have all the properties inherited from `Bid` + the `cancelDate: number;`, as long as you transform when getting the `Bid` from a database to a `CanceledBid` by adding that property.

I suppose either is fine, but this would be my preference.